### DISCUSSION TITLE
Linking directly to specific tab/accordion

### DISCUSSION THREAD

**Discussion by mattwren88** [tabpanelwidget/tabpanelwidget#16] [Q&A] [answered]
Is there a way to hyperlink directly to a specific tab or accordion?

**Comment by thierryk** [comment]
Not right now. It's something we've discussed but not built yet as there would be many challenges to address for this to work across the board/out-of-the-box.

**Reply by mattwren88** [reply]
it would be a great thing to have. This is already, by far, the best tab/accordion solution I've seen.

**Reply by DicksB** [reply]
I second the request for this feature — we've received a couple of requests from our users at the State of Missouri.

Thanks for a great library -- it's been of great help in improving the accessibility of our sites.

**Reply by thierryk** [reply]
We may look into this in a month or two...

**Reply by Knepherbird** [reply]
Just had a client request this behavior as an update to some existing pages and I had to let them know it wasn't supported. I'd like to add a plus-one for this feature.

**Reply by thierryk** [reply]
ok, we're going to work on this knowing that: 

- Devs will have to "tag" (i.e. `data-tpw-id=foo`) the widget(s) they want the script to monitor (via Query Strings).  
- The script will only keep the state of the last accordion panel that has been opened (not all of its open panels).
- Things will need to work as expected when users hit the browser's back/forward button.

One of the problem with this is that the state relates to a dom element, not to some specific content. Meaning if a URI is shared but then devs decide to reorder the content of the widget (its panels/sections), or remove some of those sections, then that URI will open the wrong panel. 
If the target panel does not exist anymore (i.e. the last panel of a widget after the number of its sections were reduced), then in that case it will fail silently.

**Reply by thierryk** [reply]
> Things will need to work as expected when users hit the browser's back/forward button.

Meaning, the users will not move through the panel states when using the browser's back/forward button as this could be a very confusing/frustrating UX.

We may offer that feature for widgets used in lieu of multiple page web sites where panels are presented as independent pages and headings as navigation links.

**Comment by thierryk** [comment, marked as answer]
This is now **available** in 2.0.0. [This link](https://tabpanelwidget.com/?tpw.demo=4&tpw.faq=0.14) should open the "And more" tab (last tab) of the first TabPanel widget on the tabpanelwidget.com page. 

Make sure to read the ["usage" section](https://tabpanelwidget.com/#usage).
